index.js

```diff
@@ -1,15 +1,57 @@
-module.exports = ({
-  user = (ctx) => `You're the following permissions: ${ctx.command.userPerms.join(', ')}`,
-  my = (ctx) => `I'm missing the following permissions: ${ctx.command.myPerms.join(', ')}`
+const humanReadable = {
+  createInvites: 'Create Invites',
+  kick: 'Kick Members',
+  ban: 'Ban Members',
+  administrator: 'Administrator',
+  manageChannels: 'Manage Channels',
+  manageGuild: 'Manage Server',
+  addReactions: 'Add Reactions',
+  auditLog: 'View Audit Log',
+  prioritySpeaker: 'Priority Speaker',
+  stream: 'Stream',
+  viewChannel: 'View Channel(s)',
+  sendMessages: 'Send Messages',
+  tts: 'Send Text-to-Speech Messages',
+  manageMessages: 'Manage Messages',
+  embed: 'Embed Links',
+  files: 'Attach Files',
+  readHistory: 'Read Message History',
+  mentionEveryone: 'Mention \@everyone, \@here, and All Roles',
+  externalEmojis: 'Use External Emoji',
+  viewInsights: 'View Server Invites',
+  connect: 'Connect (Voice)',
+  speak: 'Speak (Voice)',
+  mute: 'Mute (Voice)',
+  deafen: 'Deafen (Voice)',
+  move: 'Move (Voice)',
+  useVoiceActivity: 'Use Voice Activity',
+  nickname: 'Change Nickname',
+  manageNicknames: 'Manage Nicknames',
+  manageRoles: 'Manage Roles',
+  webhooks: 'Manage Webhooks',
+  emojis: 'Manage Emojis'
+}
+
+const exportFunc = module.exports = ({
+  humanReadable = humanReadable,
+  my = (ctx) => `I am missing the following permissions: \`${ctx.command.myPerms.filter(p => !ctx.myPerms(p)).map(p => (typeof humanReadable === 'function' ? humanReadable(ctx, p) : humanReadable[p]) || p).join('`, `')}\``,
+  user = (ctx) => `You are missing the following permissions: \`${ctx.command.userPerms.filter(p => !ctx.userPerms(p)).map(p => (typeof humanReadable === 'function' ? humanReadable(ctx, p) : humanReadable[p]) || p).join('`, `')}\``,
+  sendAsRoseError = true,
+  makeResponseAReply = false
 } = {}) => {
   return (ctx) => {
-    if (ctx.command.userPerms && !ctx.command.userPerms.every(x => ctx.hasPerms(x))) {
-      throw new Error(user(ctx))
-    }
-    if (ctx.command.myPerms && !ctx.command.myPerms.every(x => ctx.myPerms(x))) {
-      throw new Error(my(ctx))
-    }
-
-    return true
+      if (ctx.command.hasOwnProperty('myPerms') && !ctx.command.myPerms.every(x => ctx.myPerms(x))) {
+          if (sendAsRoseError && ctx.myPerms('embed')) throw new Error(my(ctx))
+          makeResponseAReply ? ctx.reply(my(ctx)) : ctx.send(my(ctx))
+          return false
+      }
+      if (ctx.command.hasOwnProperty('userPerms') && !ctx.command.userPerms.every(x => ctx.hasPerms(x))) {
+          if (sendAsRoseError) throw new Error(user(ctx))
```

Comment:
Here you do not check for `ctx.myPerms('embed')` yet you do check on line 44, perhaps one should be switched to the other. Since it is an option and throwing an error does not necessarily mean that an embed will be used, the check is probably unnecessary,

Reply:
that throw new error is handled by discord rose and its error handle only sends embeds [here](https://github.com/discord-rose/discord-rose/blob/acaf79ae5d4644b869371ab37e2e8a03e1615bab/src/structures/CommandHandler.ts#L45)

Reply:
That function should probably be changed to check for embed permissions before sending an embed. Personally I would move the check to the error handling function rather than keeping it here since developers can configure the function to handle it without an embed which would cause issues if the bot does not have the embed permissions. @jpbberry, this is up to you.

Reply:
for now i will keep the checks in place... if berry decides to have the default handler check for embed issues then it can be removed

Reply:
I think it shouldn't override the error function and should just throw even if a missing embed permissions occurs. @Million900o feel free to open a PR on the main discord-rose package to change the default error function to properly handle embed errors.